src/components/DocumentTree/model.js

```diff
@@ -0,0 +1,115 @@
+import {
+    createDocument,
+    deleteDocument,
+} from "../../services/apiManager/index.js";
+import requireNew from "../../services/requireNew.js";
+
+import { initialDocument } from "../../constants/initialData.js";
+import { replaceDocumentId } from "../../services/router.js";
+
+const validateSelf = (data) => {
+    if (typeof data.id !== "number")
+        throw new Error("문서 id가 숫자 타입이 아닙니다");
+    else if (typeof data.title !== "string")
+        throw new Error("문서 title가 문자열 타입이 아닙니다");
+    else if (Array.isArray(data.documents) === false)
+        throw new Error("문서 documents가 배열이 아닙니다");
+};
+
+export const validateDocumentNode = (nodes) => {
+    if (Array.isArray(nodes) === false) {
+        throw new Error("documentTree의 nextState가 배열이 아닙니다");
+    }
+
+    try {
+        nodes.forEach((node) => validateSelf(node));
```

Comment:
```suggestion
        nodes.forEach(validateSelf);
```
초마이너이긴 하지만, 이러면 불필요한 람다 함수 선언을 하나 줄일 수도 있을 것 같슴다 ㅎㅎㅎ